src/Umbraco.Core/Exceptions/DataOperationException.cs

```diff
@@ -1,21 +1,98 @@
 ﻿using System;
+using System.Runtime.Serialization;
 
 namespace Umbraco.Core.Exceptions
 {
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <seealso cref="System.Exception" />
+    [Obsolete("Refactor the generic type to a concrete serializable type.")]
+    [Serializable]
     internal class DataOperationException<T> : Exception
```

Comment:
This class is internal and the only type used is `MoveOperationStatusType`, so I think this could be refactored without breaking changes.

Reply:
It could be but until then we shouldn't be marking this as obsolete since it's still in use.

Reply:
I've kept the generics, but added a generic type constraint, so `T` must always be an enum 👍